src/MainWindow.vala

```diff
@@ -33,64 +32,71 @@ public class Friends.MainWindow : Hdy.ApplicationWindow {
-        sidebar_header_context.add_class ("default-decoration");
-        sidebar_header_context.add_class (Gtk.STYLE_CLASS_FLAT);
-
-        var individualview_header = new Hdy.HeaderBar () {
-            decoration_layout = ":maximize",
-            has_subtitle = false,
-            show_close_button = true
+        var individualview_header = new Gtk.WindowHandle () {
+            child = new Gtk.WindowControls (Gtk.PackType.END) {
+                halign = Gtk.Align.END
+            }
+        };
+        individualview_header.add_css_class (Granite.STYLE_CLASS_DEFAULT_DECORATION);
+        individualview_header.add_css_class ("titlebar");
+        individualview_header.add_css_class (Granite.STYLE_CLASS_FLAT);
+
+        search_entry = new Gtk.SearchEntry () {
+            margin_start = 9,
+            margin_end = 9,
+            margin_top = 3,
+            margin_bottom = 6,
+            hexpand = true,
+            placeholder_text = _("Search Friends"),
+            valign = Gtk.Align.CENTER
         };
 
-        unowned Gtk.StyleContext individualview_header_context = individualview_header.get_style_context ();
-        individualview_header_context.add_class ("default-decoration");
-        individualview_header_context.add_class (Gtk.STYLE_CLASS_FLAT);
-
-        search_entry = new Gtk.SearchEntry ();
-        search_entry.margin_start = search_entry.margin_end = 9;
-        search_entry.margin_top = 3;
-        search_entry.margin_bottom = 6;
-        search_entry.hexpand = true;
-        search_entry.placeholder_text = _("Search Friends");
-        search_entry.valign = Gtk.Align.CENTER;
-
-        listbox = new Gtk.ListBox ();
-        listbox.activate_on_single_click = true;
-        listbox.expand = true;
-        listbox.selection_mode = Gtk.SelectionMode.SINGLE;
+        listbox = new Gtk.ListBox () {
+            activate_on_single_click = true,
+            hexpand = true,
+            vexpand = true,
+            selection_mode = Gtk.SelectionMode.SINGLE,
+        };
+        listbox.add_css_class ("rich-list");
+        listbox.add_css_class ("background");
         listbox.set_filter_func (filter_function);
         listbox.set_header_func (header_function);
         listbox.set_sort_func (sort_function);
 
-        var scrolledwindow = new Gtk.ScrolledWindow (null, null);
-        scrolledwindow.add (listbox);
+        var scrolledwindow = new Gtk.ScrolledWindow () {
+            child = listbox
+        };
 
         var sidebar_grid = new Gtk.Grid ();
-        sidebar_grid.get_style_context ().add_class (Gtk.STYLE_CLASS_SIDEBAR);
         sidebar_grid.attach (sidebar_header, 0, 0);
         sidebar_grid.attach (search_entry, 0, 1);
         sidebar_grid.attach (scrolledwindow, 0, 2);
 
         var individual_view = new Friends.IndividualView ();
+        individual_view.add_css_class (Granite.STYLE_CLASS_VIEW);
 
         var individual_grid = new Gtk.Grid ();
```

Comment:
This should be on the grid instead so that it includes the header:
```suggestion

        var individual_grid = new Gtk.Grid ();
        individual_grid.add_css_class (Granite.STYLE_CLASS_VIEW);
```

Reply:
Oh yes. Thanks!